File: backend/api/routers/events.py
from fastapi import APIRouter

from backend.api.schemas.event import EventResponse
from backend.database.repository import EventRepository

router = APIRouter(
    prefix="/events",
    tags=["Events"],
)

repo = EventRepository()


@router.get(
    "",
    response_model=list[EventResponse],
)
def get_events(
    limit: int = 100,
    offset: int = 0,
    username: str | None = None,
    host: str | None = None,
    severity: str | None = None,
    event_type: str | None = None,
):

    return repo.get_events(
        limit=limit,
        offset=offset,
        username=username,
        host=host,
        severity=severity,
        event_type=event_type,
    )

File: frontend/src/api/events.ts
import { api } from "./client";

export interface EventFilters {
  username?: string;
  host?: string;
  severity?: string;
  event_type?: string;
  limit?: number;
  offset?: number;
}

export async function getEvents(filters: EventFilters) {
  const response = await api.get("/events", {
    params: filters,
  });

  return response.data;
}